src/io.github.betaseg/csbdeep_unet_train/model.py

```diff
@@ -204,50 +274,41 @@ def train(self, X, Y, validation_data, data_gen=None, epochs=None, steps_per_epo
-            for a, div_by in zip(axes, self._axes_div_by(axes)):
-                n = X.shape[ax[a]]
-                if n % div_by != 0:
-                    raise ValueError(
-                        "training images must be evenly divisible by %d along axis %s"
-                        " (which has incompatible size %d)" % (div_by, a, n)
-                    )
-
+        for x,y in zip(X,Y):
+            check_shape(x, n_dim=self.config.n_dim, n_channels=self.config.n_channel_in)
+            check_shape(y, n_dim=self.config.n_dim, n_channels=self.config.n_channel_out)
+        for x,y in zip(Xv,Yv):
+            check_shape(x, n_dim=self.config.n_dim, n_channels=self.config.n_channel_in)
+            check_shape(y, n_dim=self.config.n_dim, n_channels=self.config.n_channel_out)
+            
+        
         if epochs is None:
             epochs = self.config.train_epochs
         if steps_per_epoch is None:
             steps_per_epoch = self.config.train_steps_per_epoch
 
+        self.data_train = DataWrapper(X, Y, self.config.train_batch_size, length=epochs*steps_per_epoch, augmenter=augmenter, patch_size=self.config.patch_size)
+        self.data_val   = DataWrapper(Xv, Yv, self.config.train_batch_size, length=len(Xv), patch_size=self.config.patch_size)
+
         if not self._model_prepared:
             self.prepare_for_training()
 
+        xv, yv = next(iter(self.data_val))
+        
         if (self.config.train_tensorboard and self.basedir is not None and
-                not IS_TF_1 and not any(isinstance(cb, CARETensorBoardImage) for cb in self.callbacks)):
-            self.callbacks.append(CARETensorBoardImage(model=self.keras_model, data=validation_data,
-                                                       log_dir=str(self.logdir / 'logs' / 'images'),
-                                                       n_images=3, prob_out=self.config.probabilistic))
-
-        if data_gen is None:
-            data_gen = train.DataWrapper(X, Y, self.config.train_batch_size, length=epochs * steps_per_epoch)
+            not IS_TF_1 and not any(isinstance(cb,CARETensorBoardImage) for cb in self.callbacks)):
+            self.callbacks.append(CARETensorBoardImage(model=self.keras_model, data=(xv, yv),
+                                                       log_dir=str(self.logdir/'logs'/'images'),
+                                                       n_images=3, prob_out=self.config.probabilistic,
+                                                       **({} if care_tb_kwargs is None else care_tb_kwargs)))
 
         fit = self.keras_model.fit_generator if IS_TF_1 else self.keras_model.fit
-        history = fit(iter(data_gen), validation_data=validation_data,
+        history = fit(iter(self.data_train), validation_data=(xv, yv),
                       epochs=epochs, steps_per_epoch=steps_per_epoch,
-                      callbacks=self.callbacks, verbose=1)
+                      callbacks=self.callbacks, verbose=1, workers=4)
```

Comment:
On my Windows machine I get the following error:
 ```Your generator is NOT thread-safe.Keras requires a thread-safe generator whenuse_multiprocessing=False, workers > 1```.
So changing it to 
```
history = fit(iter(self.data_train), validation_data=(xv, yv),
                      epochs=epochs, steps_per_epoch=steps_per_epoch,
                      callbacks=self.callbacks, verbose=1, workers=1, use_multiprocessing=False)
 ```
 worked for me :)